networkx/conftest.py

```diff
@@ -248,6 +248,7 @@ def add_nx(doctest_namespace):
     "algorithms/link_analysis/hits_alg.py",
     "algorithms/link_analysis/pagerank_alg.py",
     "algorithms/node_classification.py",
+    "algorithms/polynomials.py",
```

Comment:
This shouldn't be necessary either - the polynomials module doesn't depend on `scipy`.

Reply:
When running `pytest` with `--doctest-modules`, the `nx.adjacency_matrix()` call on line 15 ends up calling `nx.to_scipy_sparse_matrix()` which requires scipy. The reason you're not picking up on it with the `default-without-scipy` CI job is likely that since sympy is not available, it probably always gets skipped.

Reply:
Ah nice catch, a sneaky transitive dependency :)

Let's go ahead and update the example to use `to_numpy_array` instead of `adjacency_matrix` - then it would need to be added to the `needs_numpy` list instead of `needs_scipy`.

Reply:
> Ah nice catch, a sneaky transitive dependency :)

You had me sweating for a minute there thinking I was wrong on both accounts!

> Let's go ahead and update the example to use `to_numpy_array` instead of `adjacency_matrix` - then it would need to be added to the `needs_numpy` list instead of `needs_scipy`.

Done.